File: app/src/main/java/com/example/movieapp/service/RetrofitInstance.java
package com.example.movieapp.service;
import retrofit2.Retrofit;
import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory;
import retrofit2.converter.gson.GsonConverterFactory;
public class RetrofitInstance {
    private static Retrofit retrofit = null;
    private static String BASE_URL = "http://www.omdbapi.com/?i=tt3896198&apikey=";
    public static MovieDataService getService(){
        if (retrofit==null){
            retrofit =  new Retrofit
                    .Builder()
                    .baseUrl(BASE_URL)
                    .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
                    .addConverterFactory(GsonConverterFactory.create())
                    .build();
        }
        return retrofit.create(MovieDataService.class);
    }

}

File: app/src/main/java/com/example/movieapp/service/RetrofitInstance.kt
package com.example.movieapp.service

import com.example.movieapp.model.Movie
import com.example.movieapp.model.MovieList

import retrofit2.Retrofit

import retrofit2.Response
import retrofit2.converter.gson.GsonConverterFactory
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

interface RetrofitInstance {


    @GET("?apikey=373c9964&s=*bal*")
    suspend fun getMovies(@Query("page")page: String) : Response<MovieList>


    @GET("?apikey=373c9964&s=a&type=movie")
    suspend fun getMoviesData() : Response<MovieList>

    @GET("?apikey=373c9964&s=*b*&type=series&page=1")
    suspend fun getSeriesData() : Response<MovieList>





    companion object {
//        private val moshi = Moshi.Builder()...
//            .add(KotlinJsonAdapterFactory())
//            .build()

      //  private var retrofit: Retrofit? = null
        private const val BASE_URL = "http://www.omdbapi.com/"
        @Synchronized
        operator fun invoke() : RetrofitInstance {
            return Retrofit.Builder()
                .addConverterFactory(GsonConverterFactory.create())
                .baseUrl(BASE_URL)
                .build()
                .create(RetrofitInstance::class.java)
        }
    }

}